src/params/index.ts

```diff
@@ -53,26 +64,103 @@ export type GetParamsByPathArgs = GetParamArgs & {recursive: boolean};
   await configureAWS(argv);
+  const ssm = new aws.SSM();
 
-  const Name = argv.path;
+  const Name = argv.withApproval ? `${argv.path}.pending` : argv.path;
   const Value = argv.value.toString();
   const Type = argv.type;
   const Overwrite = argv.overwrite;
   const KeyId = Type === 'SecureString' ? await getKMSAliasForParameter(Name) : undefined;
-  const ssm = new aws.SSM();
   const res = await ssm.putParameter({Name, Value, Type, KeyId, Overwrite}).promise();
-  return 0;
+
+  if(argv.withApproval) {
+    console.log('Parameter change is pending approval. Review change with:');
+    console.log(`  iidy --region ${argv.region} param review ${argv.path}`);
+  }
+
+  if(argv.message) {
+    await setParamTags(ssm, Name, [{Key: MESSAGE_TAG, Value: argv.message}]);
+  }
+
+  return SUCCESS;
 }
 
-async function getParamTags(path: aws.SSM.ParameterName) {
+export async function reviewParam(argv: ReviewParamArgs): Promise<number> {
+  await configureAWS(argv);
   const ssm = new aws.SSM();
+  const Name = argv.path;
+  const pendingName = `${Name}.pending`;
+  const pendingParam = await maybeFetchParam(ssm, {Name: pendingName, WithDecryption: true});
+
+  if(!_.isUndefined(pendingParam)) {
+    const currentParam = await maybeFetchParam(ssm, {Name, WithDecryption: true})
+    const pendingTags = await getParamTags(ssm, pendingName);
+    const Value = pendingParam.Value || '';
+    const currentValue = currentParam ? currentParam.Value : '<not set>';
+    const Type =  pendingParam.Type || 'SecureString';
+    const Overwrite = true;
+    const KeyId = Type === 'SecureString' ? await getKMSAliasForParameter(Name) : undefined;
+
+    console.log(`Current: ${currentValue}`);
+    console.log(`Pending: ${Value}`);
+    console.log('');
+
+    if(pendingTags[MESSAGE_TAG]) {
+      console.log(`Message: ${pendingTags[MESSAGE_TAG]}`);
+      console.log('');
+    }
+
+    const resp = await inquirer.prompt({
+      name: 'confirmed',
+      type: 'confirm',
+      default: false,
+      message: 'Would you like to approve these changes?'
+    });
+
+    if(resp.confirmed) {
+      await ssm.putParameter({Name, Value, Type, KeyId, Overwrite}).promise();
+      await ssm.deleteParameter({Name: pendingName}).promise();
+
+      const tags = _.reduce(pendingTags, (acc: aws.SSM.Tag[], Value, Key) => acc.concat({Key, Value}), []);
+      await setParamTags(ssm, Name, tags);
+      return SUCCESS;
+    } else {
+      return INTERRUPT;
+    }
+  } else {
+    console.log(`There is no pending change for parameter ${argv.path}`);
+    return FAILURE;
+  }
+}
+
+async function maybeFetchParam(ssm: aws.SSM, req: aws.SSM.GetParameterRequest): Promise<aws.SSM.Parameter|undefined> {
+  try {
+    const res = await ssm.getParameter(req).promise();
+    return res && res.Parameter;
+  } catch(e) {
+    // Return undefined if parameter does not exist
+    if(!(e.code && e.code === 'ParameterNotFound')) {
```

Comment:
This would read better if we were to explicitly return undefined on the true case and throw in the false.